src/lib/y2firewall/clients/auto.rb

```diff
@@ -198,6 +211,51 @@ def imported
       def imported?
         !!self.class.imported
       end
+
+      # Set of methods for formating all types of relations defined via
+      # has_many (@see Y2Firewall::Firewalld::Relations#has_many) in
+      # Y2Firewall::Firewalld::Zone
+      Firewalld::Zone.relations.each do |relation|
```

Comment:
Still consider that using meta-programming for this is too much.

Reply:
can be, you know I was studying a bit and this looked as a simple enough example to give it a try in practice